test/e2e/storage/testsuites/provisioning.go

```diff
@@ -73,132 +80,81 @@ func (p *provisioningTestSuite) getTestSuiteInfo() TestSuiteInfo {
 	return p.tsInfo
 }
 
-func (p *provisioningTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver TestDriver) {
+func (p *provisioningTestSuite) isTestSupported(pattern testpatterns.TestPattern, driver TestDriver) bool {
+	_, ok := driver.(DynamicPVTestDriver)
+	return ok
 }
 
-func createProvisioningTestInput(driver TestDriver, pattern testpatterns.TestPattern) (provisioningTestResource, provisioningTestInput) {
-	// Setup test resource for driver and testpattern
-	resource := provisioningTestResource{}
-	resource.setupResource(driver, pattern)
-
-	input := provisioningTestInput{
-		testCase: StorageClassTest{
-			ClaimSize:    resource.claimSize,
-			ExpectedSize: resource.claimSize,
-		},
-		cs:    driver.GetDriverInfo().Config.Framework.ClientSet,
-		pvc:   resource.pvc,
-		sc:    resource.sc,
-		dInfo: driver.GetDriverInfo(),
-	}
-
-	if driver.GetDriverInfo().Config.ClientNodeName != "" {
-		input.testCase.NodeName = driver.GetDriverInfo().Config.ClientNodeName
-	}
-
-	return resource, input
-}
-
-func (p *provisioningTestSuite) execTest(driver TestDriver, pattern testpatterns.TestPattern) {
+func (p *provisioningTestSuite) setupTest(driver TestDriver, config *TestConfig, pattern testpatterns.TestPattern) {
 	Context(getTestNameStr(p, pattern), func() {
 		var (
-			resource     provisioningTestResource
-			input        provisioningTestInput
-			needsCleanup bool
+			testCase StorageClassTest
+			pvc      *v1.PersistentVolumeClaim
+			sc       *storage.StorageClass
 		)
 
-		BeforeEach(func() {
-			needsCleanup = false
-			// Skip unsupported tests to avoid unnecessary resource initialization
-			skipUnsupportedTest(p, driver, pattern)
-			needsCleanup = true
-
-			// Create test input
-			resource, input = createProvisioningTestInput(driver, pattern)
-		})
+		Expect(pattern.VolType).To(Equal(testpatterns.DynamicPV))
```

Comment:
Expect is outside If

why do we want to rename RunTestSuite and execTest to setup*?

Reply:
Right, this `pattern.VolType` assertion must be done differently.

`RunTestSuite` and `execTest` are misnamed because they don't actual run or execute tests. They just define the test suite. The tests themselves then run later, when Ginkgo starts the test suite.

This is a subtle difference, but it is something that test developers have to understand.

Reply:
:+1: gotcha...an alternative name to "setup" could be "specify" or "define" as you've used in the comments for these functions. I took "setup" to mean purely the kind of "setting up" in `BeforeEach`es and went looking for where the `It` ended up for a bit. Might just be me...

Reply:
It changed it to use `define`.